src/react/hoc/__tests__/queries/errors.test.tsx

```diff
@@ -220,35 +221,30 @@ describe('[queries] errors', () => {
             componentDidUpdate() {
               const { props } = this;
               iteration += 1;
-              try {
-                if (iteration === 1) {
-                  // initial loading state is done, we have data
-                  expect(props.data!.loading).toBe(false);
-                  expect(props.data!.allPeople).toEqual(data.allPeople);
-                  props.setVar(2);
-                } else if (iteration === 2) {
-                  expect(props.data!.loading).toBe(true);
-                  expect(props.data!.allPeople).toBeUndefined();
-                } else if (iteration === 3) {
-                  expect(props.data!.loading).toBe(false);
-                  expect(props.data!.allPeople).toBeUndefined();
-                  // the second request had an error!
-                  expect(props.data!.error).toBeTruthy();
-                  expect(props.data!.error!.networkError).toBeTruthy();
-                  // // We need to set a timeout to ensure the unhandled rejection is swept up
-                  setTimeout(() => {
-                    try {
-                      expect(unhandled.length).toEqual(0);
-                    } catch (err) {
-                      reject(err);
-                    }
-                    done = true;
-                  });
-                } else {
-                  reject(`Too many iterations (${iteration})`);
-                }
-              } catch (err) {
-                reject(err);
+
+              if (iteration === 1) {
+                // initial loading state is done, we have data
+                expect(props.data!.loading).toBe(false);
+                expect(props.data!.allPeople).toEqual(data.allPeople);
+                props.setVar(2);
+              } else if (iteration === 2) {
+                expect(props.data!.loading).toBe(true);
+                expect(props.data!.allPeople).toBeUndefined();
+              } else if (iteration === 3) {
+                expect(props.data!.loading).toBe(false);
+                expect(props.data!.allPeople).toBeUndefined();
+                // the second request had an error!
+                expect(props.data!.error).toBeTruthy();
+                expect(props.data!.error!.networkError).toBeTruthy();
+                // // We need to set a timeout to ensure the unhandled rejection is swept up
+                setTimeout(() => {
+                  // try {
+                  expect(unhandled.length).toEqual(0);
+                  // } catch (err) {
+                  //   reject(err);
+                  // }
```

Comment:
```suggestion
                  expect(unhandled.length).toEqual(0);
```

Out of curiosity, any reason we couldn't just remove this?

Reply:
Oops, forgot to remove - thanks!